core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/WorkerInternalQueue.scala

```diff
@@ -2,16 +2,45 @@ package edu.uci.ics.amber.engine.architecture.worker.neo
 
 import java.util.concurrent.LinkedBlockingDeque
 
+import edu.uci.ics.amber.engine.architecture.worker.neo.WorkerInternalQueue.{
+  DataBatch,
+  DataPayload,
+  DummyPayload,
+  EndPayload
+}
 import edu.uci.ics.amber.engine.common.ambertag.LayerTag
 import edu.uci.ics.amber.engine.common.tuple.ITuple
 
 import scala.collection.mutable
 
+object WorkerInternalQueue {
+  // 3 kinds of data batches can be accepted by internal queue
+  trait DataBatch {
```

Comment:
Let's call this InternalQueueElement?

Reply:
or `InternalQueueEntry`?

Reply:
I have observed that you regularly use this object{ case class {}} , way of defining. Is there any specific benefit? Why not create a separate `InternalQueueElement` file?

Reply:
I still prefer element over entry because that is the term used by languages to refer to stuff inside queues/arrays etc. eg: https://www.scala-lang.org/api/current/scala/collection/mutable/Queue.html#addOne(elem:A):ArrayDeque.this.type